client/src/main/java/org/eclipse/hono/client/impl/AbstractRequestResponseClient.java

```diff
@@ -122,14 +118,42 @@ protected AbstractRequestResponseClient(
             final HonoConnection connection,
             final String tenantId) {
 
+        this(connection, tenantId, UUID.randomUUID().toString());
+    }
+
+    /**
+     * Creates a request-response client.
+     * <p>
+     * The created instance's sender link's target address is set to
+     * <em>${name}[/${tenantId}]</em> and the receiver link's source
+     * address is set to <em>${name}[/${tenantId}]/${replyId}</em>
+     * (where ${name} is the value returned by {@link #getName()}).
+     * <p>
+     * The latter address is also used as the value of the <em>reply-to</em>
+     * property of all request messages sent by this client.
+     * <p>
+     * The client will be ready to use after invoking {@link #createLinks()} or
+     * {@link #createLinks(Handler, Handler)} only.
+     *
+     * @param connection The connection to the service.
+     * @param tenantId The tenant that the client should be scoped to or {@code null} if the
+     *                 client should not be scoped to a tenant.
+     * @param replyId The replyId to use in the reply-to address.
+     * @throws NullPointerException if any of context or configuration are {@code null}.
+     */
+    protected AbstractRequestResponseClient(
+            final HonoConnection connection,
+            final String tenantId,
+            final String replyId) {
```

Comment:
I do not think that we need to provide subclasses with a way to set a specific reply-to ID. IMHO we can always simply use a UUID here.

Reply:
That would mean changing the `ApplicationClientFactory` interface, we currently have the `getOrCreateCommandClient(tenant, replyId)` method there (along with `getOrCreateCommandClient(tenant)` that uses the generated UUID).

Reply:
The motivation for the `replyId` parameter is in the javadoc of `getOrCreateCommandClient`:

> The protocol adapters need to convey an encoding of the reply-to address to the device when delivering the command. Consequently, the number of bytes transferred to the device depends on the length of the reply-to address being used. In situations where this is a major concern it might be advisable to use `getOrCreateCommandClient(String, String)` for creating a command client and provide custom (and shorter) <em>reply-to identifier</em> to be used in the reply-to address.

Reply:
Yeah, I forgot about that. So let's keep that.